app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/CheckAgencyController.scala

```diff
@@ -184,18 +186,35 @@ class CheckAgencyController @Inject()(
         }
       }
 
-    agentSubscriptionConnector.getRegistration(knownFacts.utr, knownFacts.postcode) flatMap {
-      case Some(Registration(Some(taxpayerName), isSubscribedToAgentServices)) if !isSubscribedToAgentServices =>
-        processCheckAgencyStatus(knownFacts.utr, taxpayerName, isSubscribedToAgentServices)
-      case Some(Registration(_, isSubscribedToAgentServices)) if isSubscribedToAgentServices =>
+    subscriptionService.getSubscriptionStatus(knownFacts.utr, knownFacts.postcode).flatMap {
+      case SubscriptionProcess(SubscriptionState.BrandNewSubscription, Some(registrationDetails)) =>
+        processCheckAgencyStatus(
+          knownFacts.utr,
+          registrationDetails.taxpayerName.get,
+          registrationDetails.isSubscribedToAgentServices)
+      case SubscriptionProcess(SubscriptionState.IsOnlySubscribedInETMP, Some(registrationDetails)) =>
+        agent match {
+          case hasNonEmptyEnrolments(_) =>
```

Comment:
Not suggesting this needs to  be changed, but just a thought for the future.. 
We could use a similar pattern to the  `withSubscribingAgent` function here, e.g.:
```withCleanCreds {
  // Complete partial subscription
}```

..and then within the implementation of `withCleanCreds`, it'd do a redirect to `/has-other-enrolments` if they've got unclean creds.

Reply:
can you make a tech debt ticket for this